### PR TITLE
[6.13.z] Add Capsule Provisioning test

### PR DESCRIPTION
**Pull request by Satellite-QE**
Cherrypick of PR: https://github.com/SatelliteQE/robottelo/pull/13241

Adding a test for provisioning using an external capsule.

### CODE REVIEW

**Comment by Satellite-QE** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_capsule_pxe_provisioning
provisioning: true
pod_resources_size: extra_extra_large

File: pytest_fixtures/component/provision_capsule_pxe.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    capsule_provisioning_sat,
+    module_sca_manifest_org,
+    module_lce_library,
+):
+    """
+    This fixture sets up kickstart repositories for a specific RHEL version
+    that is specified in `request.param`.
+    """
+    sat = capsule_provisioning_sat.sat
+    rhel_ver = request.param
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_capsule_pxe.py:176]:
```suggestion
    rhel_ver = request.param['rhel_version']
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_capsule_pxe_provisioning
provisioning: true
pod_resources_size: extra_extra_large

File: pytest_fixtures/component/provision_capsule_pxe.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    provisioning_type = getattr(request, 'param', '')
+    sat = module_target_sat
+    provisioning_domain_name = f"{gen_string('alpha').lower()}.foo"
+    broker_data_out = Broker().execute(
+        workflow='configure-install-sat-provisioning-rhv',
+        artifacts='last',
+        target_vlan_id=settings.provisioning.vlan_id,
+        target_host=module_capsule_configured.name,
+        provisioning_dns_zone=provisioning_domain_name,
+        sat_version='stream' if sat.is_stream else sat.version,
```
Comment (Gauravtalreja1) [inline pytest_fixtures/component/provision_capsule_pxe.py:43]:
```suggestion
        sat_version=sat.version,
```

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/api/test_provisioning.py -k test_capsule_pxe_provisioning
provisioning: true
pod_resources_size: extra_extra_large

**Review by Gauravtalreja1 — APPROVED** [review]
Ack,  PRT failures look unrelated to the changes

**Pull request merged**